helix-term/src/ui/statusline.rs

```diff
@@ -442,6 +443,21 @@ where
     write(context, title, None);
 }
 
+fn render_read_only_indicator<F>(context: &mut RenderContext, write: F)
+where
+    F: Fn(&mut RenderContext, String, Option<Style>) + Copy,
+{
+    let (title, style) = match (context.doc.readonly, context.doc.is_modified()) {
+        (true, false) => (" [readonly] ", None),
+        (true, true) => (
+            " Warning: Changing a readonly file ",
```

Comment:
I don't like the warning text here, I think it's too noisy to have help text like this in the part of the statusline with all the other widgets. I would prefer that this element only paid attention to `context.doc.readonly` and not `is_modified`